### PR TITLE
Add 0.171 release notes

### PR DESCRIPTION
**Pull request by haozhun**

### CODE REVIEW

File: presto-docs/src/main/sphinx/release/release-0.171.rst [hunk 1]
```diff
@@ -0,0 +1,27 @@
+=============
+Release 0.171
+=============
+
+General Changes
+---------------
+
+* Fix issue where final query info is not set for data definition queries.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.171.rst]: Mention that this results in leaked queries in the UI.

File: presto-docs/src/main/sphinx/release/release-0.171.rst [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+* Fix planning failure when a query contains one DISTINCT and one or more aggregations functions.
+* Fix casting from certain complex types to ``JSON`` when ``JSON`` or ``DECIMAL`` is involved.
+* Add support for ``ARRAY`` and ``ROW`` type as the 2nd argument in ``min_by`` and ``max_by``.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.171.rst]: Link to the functions. Spell out "second".

File: presto-docs/src/main/sphinx/release/release-0.171.rst [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+Web UI Changes
+--------------
+
+* Fix the threads UI (``/ui/thread``).
+
+Hive Changes
+------------
+
+* Fix deletion of files on query cancellation.
+* Add a config parameter (``hive.non-managed-table-writes-enabled``) to enable writes to non-managed (external) Hive tables.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.171.rst]:
```
* Allow writing to non-managed (external) tables using the
  ``hive.non-managed-table-writes-enabled`` config property.

Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.171.rst]: Delete "Hive" since this is redundant in the "Hive Changes" section.

Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.171.rst]:
```
* Allow writing to non-managed (external) Hive tables. This is disabled by default
  but can be enabled using ``hive.non-managed-table-writes-enabled``.

File: presto-docs/src/main/sphinx/release/release-0.171.rst [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+--------------
+
+* Fix the threads UI (``/ui/thread``).
+
+Hive Changes
+------------
+
+* Fix issue where some files are not deleted on cancellation of ``INSERT`` or ``CREATE`` queries.
+* Allow writing to non-managed (external) Hive tables. This is disabled by default
+  but can be enabled using ``hive.non-managed-table-writes-enabled``.
```
Comment (martint) [inline presto-docs/src/main/sphinx/release/release-0.171.rst]: via the xxx configuration option.

File: presto-docs/src/main/sphinx/release/release-0.171.rst [hunk 5]
```diff
@@ -0,0 +1,28 @@
+=============
+Release 0.171
+=============
+
+General Changes
+---------------
+
+* Fix planning failure when a query contains one ``DISTINCT`` and one or more aggregations functions.
```
Comment (martint) [inline presto-docs/src/main/sphinx/release/release-0.171.rst]: Fix planning regression for queries that compute a mix of distinct
  and non-distinct aggregations.

File: presto-docs/src/main/sphinx/release/release-0.171.rst [hunk 6]
```diff
@@ -0,0 +1,28 @@
+=============
+Release 0.171
+=============
+
+General Changes
+---------------
+
+* Fix planning regression for queries that compute a mix of distinct and non-distinct aggregations.
+* Fix casting from certain complex types to ``JSON`` when source type contains ``JSON`` or ``DECIMAL``.
+* Fix issue where queries are not purged after completion, and completion events are not fired
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.171.rst]: Should the comma be there? This sounds like two independent fixes. If both only apply to data definition, rephrase like

* Fix issue for data definition queries that prevented firing completion events or purging them from the coordinator's memory.

If separate, write as two separate items.

**Pull request closed without merging**